infra/horreum-infra-common/src/main/java/io/hyperfoil/tools/horreum/infra/common/resources/HorreumResource.java

```diff
@@ -0,0 +1,112 @@
+        String keycloakUrl = String.format("%s/realms/horreum", keycloakHost);
+        String horreumUrl = "http://" + networkAlias + ":8081";
+        String jdbcUrl = initArgs.get("quarkus.datasource.jdbc.url");
+        jdbcUrl = jdbcUrl.replace("localhost", networkAlias);
+
+        prop("horreum.keycloak.url", keycloakHost);
+        prop("quarkus.oidc.auth-server-url", keycloakUrl);
+        prop("quarkus.keycloak.admin-client.server-url", keycloakHost);
+        prop("quarkus.keycloak.admin-client.client-id", "horreum");
+        prop("quarkus.keycloak.admin-client.realm", "horreum");
+        prop("quarkus.keycloak.admin-client.client-secret", "**********");
+        prop("quarkus.keycloak.admin-client.grant-type", "CLIENT_CREDENTIALS");
+        prop("quarkus.oidc.token.issuer", "https://server.example.com ");
+        prop("smallrye.jwt.sign.key.location", "/privateKey.jwk");
+        prop("horreum.url", horreumUrl);
+        prop("horreum.test-mode", "true");
+        prop("horreum.privacy", "/path/to/privacy/statement/link");
+        prop("quarkus.datasource.migration.devservices.enabled", "false");
+        prop("quarkus.datasource.jdbc.url", jdbcUrl);
+        prop("quarkus.datasource.migration.jdbc.url", jdbcUrl);
+        prop("quarkus.datasource.jdbc.additional-jdbc-properties.sslmode", "require");
+        prop("amqp-host", initArgs.get("amqp.host"));
+        prop("amqp-port", initArgs.get("amqp.mapped.port"));
+        prop("amqp-username", initArgs.get("amqp-username"));
+        prop("amqp-password", initArgs.get("amqp-password"));
+        prop("amqp-reconnect-attempts", "100");
+        prop("amqp-reconnect-interval", "1000");
+        prop("quarkus.profile", "test");
+        prop("quarkus.test.profile", "test");
+        prop("horreum.bootstrap.password", "secret");
+        prop("horreum.roles.provider", "database");
+        String javaOpts = javaOptions.toString();
+
+        horreumContainer
+                .withEnv("JAVA_OPTIONS", javaOpts)
+                .withNetwork(network)
+                .withNetworkAliases(networkAlias)
+                .withCommand("/deployments/horreum.sh ");
+    }
+
+    private void prop(String key, String value) {
+        javaOptions.append(" -D").append(key).append("=").append(value).append(" ");
+    }
+
+    @Override
+    public Map<String, String> start(Optional<Network> network) {
+        if (horreumContainer == null) {
+            return Collections.emptyMap();
+        }
+        if (!network.isPresent()) {
+            horreumContainer.withNetwork(network.get());
+            horreumContainer.withNetworkAliases(networkAlias);
+        }
+
+        horreumContainer.start();
+        String horreumContainerName = horreumContainer.getContainerName().replaceAll("/", "");
+        Integer port = horreumContainer.getMappedPort(8080);
+
+        //TODO: check if this is still necessary ?
+        setProperty(HORREUM_CONTAINER_PORT, port.toString());
```

Comment:
Is this system property used anywhere?  The port is returned in a `Map` from this method and should be resolved from that Map.  I don't think we want to rely on system properties to propagate runtime variables

Reply:
Good point. I'll change how this value is retrieved.